soccer/gameplay/role_assignment.py

```diff
@@ -181,11 +181,13 @@ def assign_roles(robots, role_reqs):
     # To do our calculations though, we need to have a flat list of RoleRequirements to match to our list of robots
     # The flatten_tree() method does this and keeps track of how to rebuild the tree so we can do so at the end
     def flatten_tree(tree, path_prefix=[]):
+        valid_shell_ids = [bot.shell_id() for bot in robots]
         for key, subtree in tree.items():
             if isinstance(subtree, dict):
                 flatten_tree(subtree, path_prefix + [key])
             elif isinstance(subtree, RoleRequirements):
-                role_reqs_list.append(subtree)
+                if subtree.required_shell_id == None or subtree.required_shell_id in valid_shell_ids:
```

Comment:
since you are checking int's it dosen't matter. The difference between `==` and `is` matters only when comparing objects that have equals overrided, so if required_shell_id somehow became an object instead of an int, then it might matter. This is mostly for future cases (if we mess it up now, it can bite us in the future if we use it incorrectly).